woodwork/tests/accessor/test_series_accessor.py

```diff
@@ -0,0 +1,702 @@
+import re
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from woodwork.exceptions import ColumnNameMismatchWarning
+from woodwork.logical_types import (
+    Categorical,
+    CountryCode,
+    Datetime,
+    Double,
+    Integer,
+    NaturalLanguage,
+    Ordinal,
+    SubRegionCode,
+    ZIPCode
+)
+from woodwork.utils import import_or_none
+
+dd = import_or_none('dask.dataframe')
+ks = import_or_none('databricks.koalas')
+
+
+def xfail_dask_and_koalas(series):
+    if dd and isinstance(series, dd.Series) or ks and isinstance(series, ks.Series):
+        pytest.xfail('Dask and Koalas Accessors not yet supported.')
+
+
+def test_accessor_init(sample_series):
+    xfail_dask_and_koalas(sample_series)
+    series = sample_series.astype('category')
+    assert series.ww._schema is None
+    series.ww.init()
+    assert isinstance(series.ww._schema, dict)
+    assert series.ww.logical_type == Categorical
+    assert series.ww.semantic_tags == {'category'}
+
+
+def test_accessor_init_with_logical_type(sample_series):
+    xfail_dask_and_koalas(sample_series)
+    series = sample_series.copy().astype('string')
+    series.ww.init(logical_type=NaturalLanguage)
+    assert series.ww.logical_type == NaturalLanguage
+    assert series.ww.semantic_tags == set()
+
+    series = sample_series.copy().astype('string')
+    series.ww.init(logical_type="natural_language")
+    assert series.ww.logical_type == NaturalLanguage
+    assert series.ww.semantic_tags == set()
+
+    series = sample_series.copy().astype('string')
+    series.ww.init(logical_type="NaturalLanguage")
+    assert series.ww.logical_type == NaturalLanguage
+    assert series.ww.semantic_tags == set()
+
+
+def test_accessor_init_with_invalid_logical_type(sample_series):
+    xfail_dask_and_koalas(sample_series)
+    series = sample_series
+    error_message = "Cannot initialize Woodwork. Series dtype is incompatible with NaturalLanguage dtype. " \
+        "Try converting series dtype to string before initializing."
+    with pytest.raises(ValueError, match=error_message):
+        series.ww.init(logical_type=NaturalLanguage)
+
+
+def test_accessor_init_with_semantic_tags(sample_series):
+    xfail_dask_and_koalas(sample_series)
+    series = sample_series.astype('category')
+    semantic_tags = ['tag1', 'tag2']
+    series.ww.init(semantic_tags=semantic_tags, use_standard_tags=False)
+    assert series.ww.semantic_tags == set(semantic_tags)
+
+
+def test_accessor_init_with_name(sample_series):
+    xfail_dask_and_koalas(sample_series)
+    name = 'sample_series'
+    changed_name = 'changed_name'
+
+    use_series_name = sample_series.copy().astype('category')
+    assert use_series_name.name == name
+    assert use_series_name.ww.name == name
```

Comment:
Fair point - let me take a closer look at this.

Reply:
So, looking into this leads me to a couple thoughts. First all, I think it might be a bad idea to store the name in the column schema - a user could rename the series and then our schema name would be out of sync with the schema name. In this case, maybe the name should just always be the series name?

Second, I think we should probably raise a better error if someone tries to access a schema property before initialization as the current message of `TypeError: 'NoneType' object is not subscriptable` it not so helpful.

Reply:
Good point about the error message.

I'd be tempted to add a `__getattribute__` and disallow everything on the Accessor except for `init` when the schema isn't initialized, but things get so messy once we also start trying to grab pandas methods on it, that it feels a lot safer to use `__getattr__` instead.

Reply:
In the spirit of trying to keep each PR as simple as possible, I think I'll create a separate issue for raising a proper error when a user tries to use an accessor property without first running `init`.

Reply:
Created #572 for improving the error message.